src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstier/controller/DomainEventsListener.kt

```diff
@@ -17,20 +17,35 @@ class DomainEventsListener(
 
     @SqsListener("hmppsdomaineventsqueue", factory = "hmppsQueueContainerFactoryProxy")
     fun listen(msg: String) = runBlocking {
-        calculator.calculateTierForCrn(getCrn(msg), RecalculationSource.DomainEventRecalculation)
+        getCrn(msg)?.also {
+            calculator.calculateTierForCrn(it, RecalculationSource.DomainEventRecalculation)
+        }
     }
 
-    private fun getCrn(msg: String): String {
+    private fun getCrn(msg: String): String? {
         val (message) = objectMapper.readValue<SQSMessage>(msg)
         val domainEventMessage = objectMapper.readValue<DomainEventsMessage>(message)
-        val crn = domainEventMessage.personReference.identifiers.first { it.type == "CRN" }.value
-        log.info("Domain event received of type ${domainEventMessage.eventType} and CRN: $crn")
-        return crn
+        return if (domainEventMessage.eventType in messageTypesOfInterest) {
+            val crn = domainEventMessage.personReference.identifiers.first { it.type == "CRN" }.value
+            log.info("Domain event received of type ${domainEventMessage.eventType} and CRN: $crn")
+            crn
+        } else null
     }
 
     companion object {
-        private val log =
-            LoggerFactory.getLogger(this::class.java)
+        private val log = LoggerFactory.getLogger(this::class.java)
+
+        private val messageTypesOfInterest = listOf(
+            "assessment.summary.produced",
+            "enforcement.breach.raised",
+            "enforcement.breach.concluded",
+            "enforcement.recall.raised",
+            "enforcement.recall.concluded",
+            "probation-case.registration.added",
+            "probation-case.registration.updated",
+            "probation-case.registration.deleted",
+            "probation-case.registration.deregistered",
```

Comment:
Does this mean we'll have to maintain the list of events here as well as on the queue?  Or could we just check the message attributes to see if it's an OASys message and cancel out if it isn't `"assessment.summary.produced"` in the message?  Not the end of the world either way so happy with whatever you want to do - it's a shame OASys keeps doing things slightly differently.